test/bntest.c

```diff
@@ -3198,6 +3198,50 @@ static int file_test_run(STANZA *s)
     return 0;
 }
 
+static int test_print(void)
+{
+    int ret = 0;
+
+#ifdef OPENSSL_NO_STDIO
+    return 1;
+#else
+    BIGNUM *bn = NULL;
+    BIO *bio = NULL;
+
+    if (!TEST_ptr(bn = BN_new()))
+        return 0;
+    if (!TEST_ptr(bio = BIO_new_fp(stdout, BIO_NOCLOSE | BIO_FP_TEXT)))
+        goto err;
+
+    if (!TEST_true(BN_set_word(bn, 0x1234)))
+        goto err;
+    if (!TEST_true(BN_print_fp(stdout, bn)))
+        goto err;
+    BIO_printf(bio, "\n");
```

Comment:
Had it been me, I would rather have `bio` be a `BIO_s_mem()`, print to that using `BN_print()`, and verify the resulting string against a known string.

With the test as you currently have it, you're kind testing two things, 1) that we know how to wrap a `FILE*` into a BIO, and 2) that we have a function to print bignums...  but it's not in any way verifying that `BIO_print()` (or for that matter, `BIO_print_fp()`) actually does its job...

So, what did you actually intend to test?

Reply:
Its a coverage test for BN_print_fp().